### ISSUE TITLE
Drafts, versionning, rollbacks ...

### ISSUE THREAD

**Issue by jloosfelt** [locomotivecms/engine#524]
I open this issue to know how other users feel about this idea.

I really don't want LocomotiveCMS to become a gasworks, therefore I think that something is missing when an editor modifies a page or a model.
- Drafts would allow users to prepare a page modification and preview it before publishing it.
- Versionning would allow rollback, when I make a mistake and want to put back a page like before I modify it.

It should remain very simple for the user. I'm thinking of the history feature in Google Docs which I think is awesome.
I'm not saying it is a must-have feature, but other CMS have it and it's very useful. 

Another reason of this issue is that I really hate when a client calls me and says he needs to rollback and all I can do is restoring a backup... :)

What do you think about it ?

**Comment by ArturGajowy** [comment]
+1

When can we expect 3.0? Anytime soon? :)

@kkrol89 did your POC end up in production? :) I mean - you probably had a problem to solve so I'm interested how it all turned out. It's important for me, because I'm investigating Ruby CMS for use in our company and I think that versioning (or at least: ease of implementation of it) is quite impotant

**Comment by kkrol89** [comment]
No, this solution didn't end up in production. I haven't found solution for N+1 queries problem.
